### PR TITLE
[delphes] save trkcov information

### PR DESCRIPTION
**Pull request by vvolkl**

### CODE REVIEW

**Review by clementhelsens — APPROVED** [review]
this looks like it does the job!

**Comment by vvolkl** [comment]
Seems like in LCG_97_FCC_2 the delphes version was still 3.4.3pre01 which is too old. Updated to LCG_97_*a*_FCC_2 to use 3.4.3pre04

File: Sim/SimDelphesInterface/options/PythiaDelphes_config_IDEAtrkCov.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+metSaveTool.delphesMETArrayName = "MissingET/momentum"
+metSaveTool.delphesSHTArrayName = "ScalarHT/energy"
+metSaveTool.missingEt.Path = "met"
+
+
+## Delphes simulator -> define objects to be written out
+from Configurables import DelphesSimulation
+delphessim = DelphesSimulation()
+## Define Delphes card
+delphessim.DelphesCard = os.path.join(os.environ.get("DELPHES_DIR", ""), "cards/delphes_card_IDEAtrkcov.tcl")
```
Comment (clementhelsens) [inline Sim/SimDelphesInterface/options/PythiaDelphes_config_IDEAtrkCov.py:122]: Typo in the card name, should be
delphes_card_IDEAtrkCov.tcl

Comment (vvolkl) [inline Sim/SimDelphesInterface/options/PythiaDelphes_config_IDEAtrkCov.py:122]:
```suggestion
delphessim.DelphesCard = os.path.join(os.environ.get("DELPHES_DIR", ""), "cards/delphes_card_IDEAtrkCov.tcl")
```
yes,  thanks!

File: Sim/SimDelphesInterface/src/DelphesSaveChargedParticles.cpp [hunk 2]
```diff
@@ -50,6 +81,33 @@ StatusCode DelphesSaveChargedParticles::saveOutput(Delphes& delphes, const fcc::
     auto cand = static_cast<Candidate*>(delphesColl->At(j));
     auto particle = colParticles->create();
 
+
+    auto cov_d = CovToACTS(cand->TrackCovariance, cand->CtgTheta, cand->C);
+    auto t = colParticles_trkCov->create();
+    t.d0(1000* cand->D0);
+    t.z0(1000* cand->DZ);
+    t.phi(cand->Phi);
+    t.theta(TMath::ATan(1.0 / cand->CtgTheta) + TMath::PiOver2());
```
Comment (clementhelsens) [inline Sim/SimDelphesInterface/src/DelphesSaveChargedParticles.cpp:90]: there is a bug here, it should be:
`t.theta(TMath::ATan2(1.0,cand->CtgTheta);`		// Theta in [0, pi] range
(just checked with Bedeschi and Alcatraz)

File: Sim/SimDelphesInterface/src/DelphesSaveChargedParticles.cpp [hunk 3]
```diff
@@ -50,6 +81,33 @@ StatusCode DelphesSaveChargedParticles::saveOutput(Delphes& delphes, const fcc::
     auto cand = static_cast<Candidate*>(delphesColl->At(j));
     auto particle = colParticles->create();
 
+
+    auto cov_d = CovToACTS(cand->TrackCovariance, cand->CtgTheta, cand->C);
+    auto t = colParticles_trkCov->create();
+    t.d0(1000* cand->D0);
+    t.z0(1000* cand->DZ);
+    t.phi(cand->Phi);
+   t.theta(TMath::ATan2(1.0,cand->CtgTheta);
```
Comment (vvolkl) [inline Sim/SimDelphesInterface/src/DelphesSaveChargedParticles.cpp:90]:
```suggestion
   t.theta(TMath::ATan2(1.0, cand->CtgTheta));
```

**Pull request merged**